intelmq/bots/experts/url/expert.py

```diff
@@ -0,0 +1,135 @@
+# SPDX-FileCopyrightText: 2023 Filip Pokorný
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+from typing import Optional, List
+from urllib.parse import urlparse
+
+from intelmq.lib.bot import ExpertBot
+from intelmq.lib.harmonization import IPAddress, FQDN
+
+URLSCHEMES = {
+    "ftp": {
+        "default_port": 21,
+        "transport_protocol": "tcp"
+    },
+    "http": {
+        "default_port": 80,
+        "transport_protocol": "tcp"
+    },
+    "https": {
+        "default_port": 443,
+        "transport_protocol": "tcp",
+        "application_protocol": "http"
```

Comment:
This is true if we ignore HTTP/3 (which uses QUIC as transport protocol). And HTTP/3 is already used by many popular sites, but _probably_ not the ones we are primarily looking for. If we consider HTTP/3 irrelevant now and assume `https://` means HTTP/1.1 and HTTP/2 (and thus, TCP), at what point in the future do we drop this assumption?

Reply:
I don't want to start a debate on principles. However, I think we must not add assumptions to events which might not be true (even if the chance is > 50%), as that can lead to inaccurate data and investigations that come to a dead end (e.g. because the `tcpdump` filter used the wrong filter).

Reply:
This is a very good remark.

At what point? Probably when HTTP/2 (and older) is considered so obsolete that only HTTP/3 (or higher at that point) is generally used. Which is something that I don't see happening in the next decade (maybe more)? There is still going to be a plenty of clients that "don't speak QUIC" for a long long time and the world will keep older HTTP available for them. At least this is my assumption.

Reply:
Many clients support it and some websites, too (including Cloudflare).
https://w3techs.com/technologies/details/ce-http3 does not have many details, but it says that a quarter of the top million websites support it. I did not find statistics on how significant the share of total traffic is.

Reply:
But how many websites support HTTP/3 **only**? Most likely less than 1%?

As long as they support HTTP/2 (and older) then assuming `"transport_protocol": "tcp"` is not wrong (just possibly incomplete).

Reply:
Don't get me wrong, I do not disagree with you. I just think the chance is not just "> 50%" but more likely ">99%". But I agree that it is not 100%.

I will amend the code.